lib/evaluate/intrinsics.cc

```diff
@@ -1497,8 +1497,9 @@ static bool ApplySpecificChecks(
     if (const auto &arg{call.arguments[0]}) {
       if (const auto *expr{arg->UnwrapExpr()}) {
         if (const Symbol * symbol{GetLastSymbol(*expr)}) {
-          ok = symbol->has<semantics::ObjectEntityDetails>() &&
-              symbol->attrs().test(semantics::Attr::ALLOCATABLE);
+          const Symbol &resolved{ResolveAssociations(*symbol)};
+          ok = resolved.has<semantics::ObjectEntityDetails>() &&
+              resolved.attrs().test(semantics::Attr::ALLOCATABLE);
```

Comment:
Technically,   `Within an ASSOCIATE, CHANGE TEAM, or SELECT TYPE construct, [...] The associating entity does not have the ALLOCATABLE or POINTER attributes`
Which only leaves SELECT RANK where it would be legal to call on an associated entity.
However, I am not sure it is very harmful to allow the use of ALLOCATED in all associations kind.
Should you also call `ResolveAssociations` for `ASSOCIATED` below ?

Reply:
Good point.  I'll revert the test to a simple check for the ALLOCATABLE attribute on the original symbol, and trust name resolution to get it right in the case of SELECT RANK.